quiche/src/recovery/congestion/cwnd_unchecked.rs

```diff
@@ -0,0 +1,119 @@
+use std::time::Instant;
+
+use crate::recovery;
+
+use crate::recovery::congestion::Acked;
+use crate::recovery::congestion::UNCHECKED_WINDOW;
+use crate::recovery::rtt::RttStats;
+use crate::recovery::Sent;
+
+use super::Congestion;
+use super::CongestionControlOps;
+
+pub(crate) static CONGESTION_WINDOW_UNCHECKED: CongestionControlOps =
+    CongestionControlOps {
+        on_init,
+        on_packet_sent,
+        on_packets_acked,
+        congestion_event,
+        checkpoint,
+        rollback,
+        has_custom_pacing,
+        #[cfg(feature = "qlog")]
+        state_str,
+        debug_fmt,
+    };
+
+pub fn on_init(r: &mut Congestion) {
+    r.congestion_window = UNCHECKED_WINDOW;
+}
+
+pub fn on_packet_sent(
+    _r: &mut Congestion, _sent_bytes: usize, _bytes_in_flight: usize,
+    _now: Instant,
+) {
+}
+
+fn on_packets_acked(
+    r: &mut Congestion, _bytes_in_flight: usize, packets: &mut Vec<Acked>,
+    now: Instant, rtt_stats: &RttStats,
+) {
+    for pkt in packets.drain(..) {
+        on_packet_acked(r, &pkt, now, rtt_stats);
+    }
+}
+
+fn on_packet_acked(
+    r: &mut Congestion, packet: &Acked, now: Instant, rtt_stats: &RttStats,
+) {
+    if r.in_congestion_recovery(packet.time_sent) {
+        return;
+    }
+
+    if r.app_limited {
+        return;
+    }
+
+    if r.congestion_window < r.ssthresh.get() {
+        // In Slow start, bytes_acked_sl is used for counting
+        // acknowledged bytes.
+        r.bytes_acked_sl += packet.size;
+
+        r.congestion_window = UNCHECKED_WINDOW;
+
+        if r.hystart.on_packet_acked(packet, rtt_stats.latest_rtt, now) {
+            // Exit to congestion avoidance if CSS ends.
+            r.ssthresh.update(r.congestion_window, true);
+        }
+    } else {
+        // Congestion avoidance.
+        r.bytes_acked_ca += packet.size;
+
+        if r.bytes_acked_ca >= r.congestion_window {
+            r.bytes_acked_ca -= r.congestion_window;
```

Comment:
I guess this is reachable now that `r.congestion_window` is only 2GB or 16GB.

I don't have a good intuition for why this code is needed though.  Although I'm fine with this code mostly going in as is, I'ld like that we put some effort into simplifying this stuff further and figuring out what parts are actually needed and which aren't.